### DISCUSSION TITLE
How to process response headers with apiClient

### DISCUSSION THREAD

**Discussion by TB-Koma** [Himenon/openapi-typescript-code-generator#78] [Q&A] [answered]
Hello,

Thank you for developing this awesome generator.
Now I want to get values of not only response content but response headers like "X-Total-Count" by using apiClient.
But apiClient generated by default setup lacks logics of parsing response headers which the original OpenAPI schema defines. 
Could you show me examples how to setup CodeGenerator properly if possible?

My current code:

```TypeScript
const codeGenerator = new CodeGenerator(outFile)
codeGenerator.validateOpenApiSchema()

const apiClientGeneratorTemplate: Types.CodeGenerator.CustomGenerator<Templates.ApiClient.Option> =
{
  generator: Templates.ApiClient.generator,
  option: {},
}

let code = codeGenerator.generateTypeDefinition([
  codeGenerator.getAdditionalTypeDefinitionCustomCodeGenerator(),
  apiClientGeneratorTemplate,
])
```

**Comment by Himenon** [comment, marked as answer]
Thanks @TB-Koma  for using the library.

There are two ways to make the Response Header be accepted directly as the return value of a Method.

The first way is to extend `RequestOption` after code generation. I wrote a sample code in this Pull Request.

 https://github.com/Himenon/openapi-typescript-code-generator/pull/79/files#diff-a34ad3f9712a13614328783b1b59a378140f85cef97da12a840f4416096cdcdaR6-R58

This method is the one you can try right now.


The second method is to create your own code template, although it is more time-consuming.
If you are familiar with AST, you can write code to receive the Header directly from the Method return value by modifying it here.

 https://github.com/Himenon/openapi-typescript-code-generator/blob/main/src/code-templates/api-client/ApiClientClass/Method.ts#L38-L94

My personal recommendation is that the first method is easier.

**Reply by TB-Koma** [reply]
I totally overlooked that this issue can be solved easily by injecting a callback function like the first method
and I managed to implement a client code without any problems.

@Himenon  I appreciate your help!
